### PR TITLE
[test] Update tests for parity with the new driver

### PR DESCRIPTION
**Pull request by hamishknight**
Update tests such that they run with both the old and new driver in preparation for switching over to the new driver.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by QuietMisdreavus** [comment]
SymbolGraphGen test update looks good.

How do you force running tests with the new driver? I'm interested to know so i can double-check during the transition period.

**Comment by hamishknight** [comment]
@QuietMisdreavus You can force the new driver by setting `SWIFT_FORCE_TEST_NEW_DRIVER=1` in your environment

**Review by artemcm — APPROVED** [review]

**Comment by hamishknight** [comment]
https://github.com/swiftlang/swift-syntax/pull/2858

@swift-ci please test

File: test/SILOptimizer/default-cmo.swift [hunk 1]
```diff
@@ -5,14 +5,14 @@
 // RUN: %target-build-swift -O -wmo -Xfrontend -enable-default-cmo -parse-as-library -emit-module -emit-module-path=%t/Module.swiftmodule -module-name=Module -package-name Pkg -I%t -I%S/Inputs/cross-module %S/Inputs/cross-module/default-module.swift -c -o %t/module.o
 // RUN: %target-build-swift -O -wmo -Xfrontend -enable-default-cmo -parse-as-library -emit-tbd -emit-tbd-path %t/ModuleTBD.tbd -emit-module -emit-module-path=%t/ModuleTBD.swiftmodule -module-name=ModuleTBD -package-name Pkg -I%t -I%S/Inputs/cross-module %S/Inputs/cross-module/default-module.swift -c -o %t/moduletbd.o -Xfrontend -tbd-install_name -Xfrontend module
 
-// RUN: %target-build-swift -O -wmo -module-name=Main -package-name Pkg -I%t -I%S/Inputs/cross-module %s -emit-sil | %FileCheck %s
+// RUN: %target-build-swift -Xfrontend -enable-default-cmo -O -wmo -module-name=Main -package-name Pkg -I%t -I%S/Inputs/cross-module %s -emit-sil | %FileCheck %s
```
[the remark this answers is not in the document]
Comment (hamishknight) [reply test/SILOptimizer/default-cmo.swift:8]: The test suite currently uses the old driver, where it's disabled by default

**Comment by hamishknight** [comment]
@swift-ci please test Linux

**Pull request merged**